src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
+      console.error(`Exception occurred while fetching shop list : ${e}`);
+      return [];
+    }
+  }
+}
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+
+// Calculate distance of each shop from the given coordinate
+export default async function getNearestShops(position) {
+  const shopData = await getShopData();
+
+  if (shopData.length > 0) {
+    let distancelist = [];
```

Comment:
Seeing that you create an empty array, push into it several times and sort it; are you familiar with the concept of side effects? Can you think of an implementation strategy that would enable you to do all these operations?

Reply:
I am not sure that I know the concept of side effects in Javascript. But as far as I understand from your question, there is a simple way to do this without creating an empty array at first and sorting it somewhere else. Maybe I can come up with a better solution if I can think more about it but currently, I don't have any other idea to implement this better.